Fix some problems with Vue.js account creation and login

LefterisJP opened the pull request:
## What does the PR do

Fix #616 

- New Account: Check password confirmation matches provided password
- Catch non-2XX responses for account login and show error to user 

## Feedback needed

@kelsos I need help here.

I discovered that Axios, at least with the current settings you have treats any non `2XX` response as an error. So catching errors from the API needs to be in the `        .catch(error => {` part of the code.

Saw a gist explaining how to do that [here](https://gist.github.com/fgilio/230ccd514e9381fafa51608fcf137253).

But that seems like a lot of work to put such logic in each endpoint of Rotki in `rotkehlchen-api.ts`.

Especially since almost all of the endpoints do return non 2XX codes which should be process-able by the front-end. Can't we generalize this somehow? To check the status code if it matches the ones expected by the endpoint from the docs. If it does, process it and if there is no result get the message and show it to the user as an error. Else it's  probably a `500` error and then the user should check the backend logs.

Anyway as you can see from commit https://github.com/rotki/rotki/commit/a47a54107c0e4625d2a23e94b106a9a33dea42fb I actually did something close to this logic only for the login endpoint but something is wrong. The first time I tried to login I did get the `'message'` text shown to me as I wanted: (Check also the console)

![2020-01-07-000504_1330x610_scrot](https://user-images.githubusercontent.com/1658405/71855854-82989000-30e2-11ea-97bc-20abe8a704dd.png)


The second time I tried I got a 409 error and checking the console I see different endpoints being queried. Why?

![2020-01-07-000517_1552x624_scrot](https://user-images.githubusercontent.com/1658405/71855902-9d6b0480-30e2-11ea-93a0-d22fc52f56b0.png)

kelsos commented:
So other than creating a handler/function to handle the errors I cannot think of anything else, this function has to go to each different call basically. 

About the other errors we probably need to check what happens in the app after the wrong password

LefterisJP commented:
@kelsos Can't this handler be generic enough to be used in all of the calls?

kelsos commented:
Sure and I need to check if Axios supports interceptors. I believe it does but I have to verify. If interceptors are supported then the error handling logic can go there and we won't have to call it at every single call.

Second I found out why the settings/exchanges call fail. GET `/api/1//users` reports the user that had the wrong password as `loggedin`. If you are already logged in the login api will return an error, so there is a piece of logic there that will check the status of the user attempting to login, and if the user is logged it will try to fetch the settings/exchanges.

LefterisJP commented:
> Second I found out why the settings/exchanges call fail. GET /api/1//users reports the user that had the wrong password as loggedin

So that sounds like a backend error. I will check it.

LefterisJP commented:
Also delete `sync_approval` from create_user endpoint. Check if it's used but I believe it's not used there.

LefterisJP commented:
> So that sounds like a backend error. I will check it.

It was a backend error. Fixed here and added tests: https://github.com/rotki/rotki/pull/617/commits/64e6e2046f6947126f866315d9d494c1a616b15c

LefterisJP commented on electron-app/src/components/CreateAccount.vue:
```diff
@@ -93,7 +93,10 @@ export default class CreateAccount extends Vue {
-    (v: string) => !!v || 'Please provide a password confirmation'
+    (v: string) => !!v || 'Please provide a password confirmation',
+    (v: string) =>
+      v == this.password ||
```
@kelsos this does not work. It is always false. How can I properly compare the password confirm to the password field for validation purposes in the UI?

Also can there be a unit test for this? Sounds unit-testable.

kelsos replied:
Yes, the problem is that the rule is getting passed the initial string reference in the beginning, and then when the password changes the reference changes. This means that for the rule `this.password` is always an empty string (a reference to the initial string). We could probably work around that by adding a computed property.

LefterisJP replied:
What is a computed property? How does that work?

LefterisJP commented on electron-app/src/services/rotkehlchen-api.ts:
```diff
@@ -59,12 +59,26 @@ export class RotkehlchenApi {
-        .patch<ActionResult<boolean>>(`/users/${username}`, {
-          action: 'logout'
-        })
+        .patch<ActionResult<boolean>>(
```
It's a big diff but essentially just added the status code validators in all endpoints. Have a look and tell me if it makes sense @kelsos

LefterisJP commented:
@kelsos so you can have a look now. Please also advice on the password confirmation [comment](https://github.com/rotki/rotki/pull/617#discussion_r365477782).

kelsos commented on electron-app/src/services/rotkehlchen-api.ts:
```diff
@@ -82,7 +96,11 @@ export class RotkehlchenApi {
+        .get<ActionResult<PeriodicClientQueryResult>>('/periodic/', {
+          validateStatus: function(status) {
+            return status == 200 || status == 409;
+          }
```
nitpick: can we make these functions that are inline to arrow functions?
```typescript
          validateStatus: (status) => status == 200 || status == 409;
```
or 

```typescript
          validateStatus: (status) => {
            return status == 200 || status == 409;
          }
```

kelsos commented on electron-app/src/services/rotkehlchen-api.ts:
```diff
@@ -230,12 +273,20 @@ export class RotkehlchenApi {
     });
   }
 
+  private validate_status_put_settings(status: number) {
```
nitpick: Since in TypeScript the functions are usually in camelcase can we make sure that we follow this for function naming? (I will probably enforce it by eslint in the future, and we can add exceptions in a few models that are coming from JSON where this makes sense) :)

LefterisJP commented:
For styling comments @kelsos please do them yourself and enforce them with linting.
The most important thing I need help with is the password confirmation validation. Did you not see my comment?

LefterisJP commented:
Okay so with your change @kelsos the password confirmation validation works. Tests are needed but they can go in another PR.
